gtsam/hybrid/HybridBayesTree.h

```diff
@@ -0,0 +1,102 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    HybridBayesTree.h
+ * @brief   Hybrid Bayes Tree, the result of eliminating a HybridJunctionTree
+ * @date Mar 11, 2022
+ * @author  Fan Jiang
+ */
+
+#pragma once
+
+#include <gtsam/hybrid/HybridBayesNet.h>
+#include <gtsam/hybrid/HybridFactorGraph.h>
+#include <gtsam/inference/BayesTree.h>
+#include <gtsam/inference/BayesTreeCliqueBase.h>
+#include <gtsam/inference/Conditional.h>
+
+#include <string>
+
+namespace gtsam {
+
+// Forward declarations
+class HybridConditional;
+class VectorValues;
+
+/* ************************************************************************* */
+/** A clique in a HybridBayesTree
+ * which is a HybridConditional internally.
+ */
+class GTSAM_EXPORT HybridBayesTreeClique
+    : public BayesTreeCliqueBase<HybridBayesTreeClique, HybridFactorGraph> {
+ public:
+  typedef HybridBayesTreeClique This;
+  typedef BayesTreeCliqueBase<HybridBayesTreeClique, HybridFactorGraph> Base;
+  typedef boost::shared_ptr<This> shared_ptr;
+  typedef boost::weak_ptr<This> weak_ptr;
+  HybridBayesTreeClique() {}
+  virtual ~HybridBayesTreeClique() {}
+  HybridBayesTreeClique(const boost::shared_ptr<HybridConditional>& conditional)
+      : Base(conditional) {}
+};
+
+/* ************************************************************************* */
+/** A Bayes tree representing a Hybrid density */
+class GTSAM_EXPORT HybridBayesTree : public BayesTree<HybridBayesTreeClique> {
+ private:
+  typedef BayesTree<HybridBayesTreeClique> Base;
+
+ public:
+  typedef HybridBayesTree This;
+  typedef boost::shared_ptr<This> shared_ptr;
+
+  /// @name Standard interface
+  /// @{
+  /** Default constructor, creates an empty Bayes tree */
+  HybridBayesTree() = default;
+
+  /** Check equality */
+  bool equals(const This& other, double tol = 1e-9) const;
+
+  /// @}
+};
+
+/* This does special stuff for the hybrid case */
+template <class CLIQUE>
+class BayesTreeOrphanWrapper<
+    CLIQUE, typename std::enable_if<
+                boost::is_same<CLIQUE, HybridBayesTreeClique>::value> >
+    : public CLIQUE::ConditionalType {
+ public:
+  typedef CLIQUE CliqueType;
+  typedef typename CLIQUE::ConditionalType Base;
+
+  boost::shared_ptr<CliqueType> clique;
+
+  BayesTreeOrphanWrapper(const boost::shared_ptr<CliqueType>& clique)
```

Comment:
Can't move the implementation since the class is templated :(

Reply:
Yes, this is deliberate because `BayesTreeOrphanWrapper` needs special implementation specific to hybrid, which is guarded by `std::enable_if<
                 boost::is_same<CLIQUE, HybridBayesTreeClique>::value>`